pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -1056,6 +1056,22 @@ def test_dt64arr_add_sub_parr(
         )
         assert_invalid_addsub_type(dtarr, parr, msg)
 
+    def test_timestamp_and_time_dtype_raises(self, box_with_array):
+        # https://github.com/pandas-dev/pandas/issues/10329
+        obj1 = pd.date_range("2012-01-01", periods=3)
+        obj2 = [time(i, i, i) for i in range(3)]
+
+        obj1 = tm.box_expected(obj1, box_with_array)
+        obj2 = tm.box_expected(obj2, box_with_array)
+
+        msg = (
+            "unsupported operand type\(s\) for -: "
+            "'(Timestamp|DatetimeArray)' and 'datetime.time'"
+        )
+
+        with pytest.raises(TypeError, match=msg):
+            obj1 - obj2
```

Comment:
the reversed operation obj2 - obj1 should also raise right?  How about the addition ops?